stl/src/cthread.cpp

```diff
@@ -99,9 +99,16 @@ _Thrd_id_t _Thrd_id() { // return unique id for current thread
 }
 
 unsigned int _Thrd_hardware_concurrency() { // return number of processors
-    SYSTEM_INFO info;
-    GetNativeSystemInfo(&info);
-    return info.dwNumberOfProcessors;
+    DWORD_PTR process_affinity;
+    DWORD_PTR system_affinity;
+    if (GetProcessAffinityMask(GetCurrentProcess(), &process_affinity, &system_affinity))
+#ifdef _WIN64
+        return __builtin_popcountll(process_affinity);
+#else

```

Comment:
I think that either

* both the `#else` and `#endif` should be commented, or
* neither should be commented.

Dropping the comment is fine. (#351 remains active.)

Reply:
I dropped it. I did not mean to put it, it was IDE assistance; for one-liners they are unnecessary I think.